docs/DeveloperGuide.md

```diff
@@ -527,76 +525,200 @@ Use case ends
+   Use case ends.
+    
+**Use case: UC06 Tag food**
 
 **MSS**
+1. User requests to list food (UC01).
+2. McGymmy shows a list of food.
+3. User request to tag a specific food on the list.
+4. McGymmy updates the food with the given tag.<br>
+Use case ends.
 
-1. User creates a macro to execute two 'add' commands in sequence.
-2. McGymmy adds the macro to the list of available commands.
-3. User uses the newly added macro command.
-4. McGymmy executes the two commands consecutively.
+**Extensions**
+- 2a. The list is empty.<br>
+   Use case ends.
+    
+- 3a. The given index is invalid.<br>
+   - 3a1. McGymmy shows an error message.<br>
+   Use case resumes at step 2.
 
+- 3b. The given tag already exists for the selected food.<br>
+   - 3a1. McGymmy shows an error message.<br>
+   Use case resumes at step 2.
+
+- 3c. The format of tag method is invalid.
+   - 3c1. McGymmy shows an error message.<br>
+   Use case ends.
+    
+**Use case: UC07 Untag food**
+
+**MSS**
+1. User requests to list food (UC01).
+2. McGymmy shows a list of food.
+3. User request to remove tag from a specific food on the list.
+4. McGymmy removes the tag from the food.<br>
 Use case ends
 
 **Extensions**
+- 2a. The list is empty.<br>
+   Use case ends.
 
- - 1a. The format of the macro is invalid.
-    - 1a1. McGymmy shows an error message.
+- 3a. The given index is invalid.<br>
+   - 3a1. McGymmy shows an error message.<br>
+   Use case resumes at step 2.
+
+- 3b. The given tag does not exist for the selected food.<br>
+   - 3a1. McGymmy shows an error message.<br>
+   Use case resumes at step 2.
+
+- 3c. The format of untag method is invalid.
+   - 3c1. McGymmy shows an error message.<br>
+   Use case ends.
+    
+**Use case: UC08 Find**
+
+**MSS**
+1. User requests to find (filter) food items based on certain criteria.
+2. McGymmy shows an updated list based on the filter criteria.
+
+Use case ends.
+
+**Extensions**
+- 1a. The format of find method is invalid.
+   - 1a1. McGymmy shows an error message.
 
 Use case ends.
+    
+**Use case: UC09 Clear**
 
- - 4a. One of the executed commands encounter an error.
-    - 4a1. McGymmy shows the error message from that command, and displays the commands that successfully executed, and the commands that have yet to execute.
+**MSS**
+1. User filters list (UC06).
+2. User requests to clear the displayed list.
+3. McGymmy deletes all food items in the current displayed list.
 
 Use case ends.
 
-**Use case: UC07 Undo the last command**
+**Extensions**
+- 1a. The list is empty.<br>
+    
+Use case ends.
+
+**Use case: UC10 Import a McGymmy data file**
 
 **MSS**
 
-1. User requests to undo the last command.
-2. McGymmy successfully undo the last command.
+1. User imports McGymmy file to McGymmy.
+2. McGymmy overrides the old files with the new McGymmy files.<br>
+Use case ends.
 
+**Extensions**
+ - 1a. The format for the save file is invalid.
+    - 1a1. McGymmy shows an error message.
+    Use case ends.
```

Comment:
missing break line before this line
![image](https://user-images.githubusercontent.com/58925632/98467858-a650c900-2212-11eb-8062-727efc2cdc66.png)